server/routes/room-directory-routes.js

```diff
@@ -31,19 +32,31 @@ router.get(
     const paginationToken = req.query.page;
     const searchTerm = req.query.search;
 
-    const { rooms, nextPaginationToken, prevPaginationToken } = await fetchPublicRooms(
-      matrixAccessToken,
-      {
-        //server: TODO,
-        searchTerm,
-        paginationToken,
-        // It would be good to grab more rooms than we display in case we need
-        // to filter any out but then the pagination tokens with the homeserver
-        // will be out of sync. XXX: It would be better if we could just filter
-        // `/publicRooms` directly via the API (needs MSC).
-        limit: 9,
-      }
-    );
+    // It would be good to grab more rooms than we display in case we need
+    // to filter any out but then the pagination tokens with the homeserver
+    // will be out of sync. XXX: It would be better if we could just filter
+    // `/publicRooms` directly via the API (needs MSC).
+    const limit = 9;
+
+    let rooms;
+    let nextPaginationToken;
+    let prevPaginationToken;
+    try {
+      ({ rooms, nextPaginationToken, prevPaginationToken } = await fetchPublicRooms(
+        matrixAccessToken,
+        {
+          //server: TODO,
+          searchTerm,
+          paginationToken,
+          limit,
+        }
+      ));
+    } catch (err) {
+      throw new RethrownError(
+        `Unable to fetch rooms from room directory (homeserver=${matrixServerUrl})\n    searchTerm=${searchTerm}, paginationToken=${paginationToken}, limit=${limit}`,
+        err
+      );
```

Comment:
This will still end up as a 500 error to the client but at least it's explained better.

I can't think of a better HTTP status code to return in this situation though 🤔

We could return a room directory with an empty list of rooms (hopefully with the error explained).

Reply:
We can iterate on this later ⏩